controllers/nodemaintenance_controller.go

```diff
@@ -241,7 +235,7 @@ func onPodDeletedOrEvicted(pod *corev1.Pod, usingEviction bool) {
 	klog.Info(msg)
 }
 
-func initDrainer(r *NodeMaintenanceReconciler, config *rest.Config) error {
+func initDrainer(r *NodeMaintenanceReconciler, config *rest.Config, ctx context.Context) error {
```

Comment:
passing a TODO context from main does not provide any value over using the BACKGROUND context as before in this function, does it? It's a bit ugly that the Helper struct needs the context, would be better if the actual functions of the helper take a context as parameter, but that's not on us. I think the cleanest solution is to create a Helper for each reconcile, when we need it, that's when we have the context. Since the Helper struct isn't doing anything when being created, the overhead is low. WDYT?

Reply:
> we will only set the ctx on every reconcile run (or before every usage of the Helper (e.g., r.drainer))?

Think about about it... having a single helper on the reconciler, but modifying a field on every reconcile. And having the option to run concurrent reconciles... (https://github.com/medik8s/node-healthcheck-operator/blob/d646bf09ee114eb00c70775e56589cb850cfaba3/vendor/sigs.k8s.io/controller-runtime/pkg/config/controller.go#L35). Does that sound like a good idea? 🙂

Reply:
With concurrency, it is indeed not the right "move" of creating once and modifying it each time. Creating a new instance each time sounds good to me.